### PR TITLE
Telemetry UI Updates

### PR DESCRIPTION
**Pull request by JeremyQuartech**
## Description of Changes

- Updated styling of Add telemetry device dialog
- Changed edit telemetry details to render each device as a card
- Minor changes to Individual animals nomenclature
- Styling improvements on individual animals table
- Styling improvements on general animal form
- Minor bug fix in /deployments api

### CODE REVIEW

File: app/src/features/surveys/view/survey-animals/form-sections/GeneralAnimalForm.tsx [hunk 1]
```diff
@@ -24,58 +23,65 @@ const GeneralAnimalForm = () => {
...
           <HelpButtonTooltip content={SurveyAnimalsI18N.taxonHelp}>
             <CbSelectField
               name={getAnimalFieldName<IAnimalGeneral>(name, 'taxon_id')}
               controlProps={{
                 required: isRequiredInSchema(AnimalGeneralSchema, 'taxon_id'),
-                size: 'small',
                 disabled: !!values.collectionUnits.length
               }}
-              label={'Taxon'}
+              label={'Species'}
```
Comment (GrahamS-Quartech) [inline app/src/features/surveys/view/survey-animals/form-sections/GeneralAnimalForm.tsx:47]: I think Taxon is more accurate for this field but if this change comes from Macgregor then all good

File: api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments.ts [hunk 2]
```diff
@@ -100,8 +100,8 @@ POST.apiDoc = {
       content: {
         'application/json': {
           schema: {
-            title: 'Number of rows affected',
-            type: 'number'
+            title: 'Deployment response object',
+            type: 'object'
```
Comment (NickPhura) [inline api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments.ts]: nit: specify that this is an object with 1 field "message"

Same for the other response below

File: app/src/components/buttons/HelpButtonTooltip.tsx [hunk 3]
```diff
@@ -16,28 +16,52 @@ interface HelpButtonTooltipProps {
  * @return {*}
  */
 
-const HelpButtonTooltip = ({ content, children }: HelpButtonTooltipProps) => {
+const HelpButtonTooltip = ({ content, children }: HelpButtonTooltipProps): any => {
```
Comment (NickPhura) [inline app/src/components/buttons/HelpButtonTooltip.tsx]: We should avoid the "any" here.

(If this is applicable/helpful, there is a `PropsWithChildren<T>` type you can use, to add children to any existing type.

**Comment by NickPhura** [comment]
I add a single animal to a survey.

When the survey page loads (just loading the survey page, not clicking anything) I'm getting the following console error:

```
Warning: Encountered two children with the same key, `taxon`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
GridColumnHeadersInner@http://localhost:7107/static/js/bundle.js:128052:9
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
GridColumnHeaders@http://localhost:7107/static/js/bundle.js:127158:9
GridColumnsHeaders@http://localhost:7107/static/js/bundle.js:123752:9
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
./node_modules/@mui/x-data-grid/components/containers/GridMainContainer.js/GridMainContainer<@http://localhost:7107/static/js/bundle.js:128715:100
GridBody@http://localhost:7107/static/js/bundle.js:124781:7
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
GridRoot@http://localhost:7107/static/js/bundle.js:128866:100
GridContextProvider@http://localhost:7107/static/js/bundle.js:133348:29
DataGrid@http://localhost:7107/static/js/bundle.js:121736:84
CustomDataGrid@http://localhost:7107/static/js/bundle.js:17329:19
SurveyAnimalsTable@http://localhost:7107/static/js/bundle.js:49264:28
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
SurveyAnimals@http://localhost:7107/static/js/bundle.js:44226:81
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Paper@http://localhost:7107/static/js/bundle.js:103888:83
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Grid@http://localhost:7107/static/js/bundle.js:97032:87
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Grid@http://localhost:7107/static/js/bundle.js:97032:87
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Container@http://localhost:7107/static/js/bundle.js:117455:32
SurveyPage@http://localhost:7107/static/js/bundle.js:46042:73
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
ProjectsLayout@http://localhost:7107/static/js/bundle.js:60594:19
Route@http://localhost:7107/static/js/bundle.js:274891:29
RouteWithTitle@http://localhost:7107/static/js/bundle.js:63199:7
Switch@http://localhost:7107/static/js/bundle.js:275060:29
SurveyRouter
ObservationsContextProvider@http://localhost:7107/static/js/bundle.js:19543:93
SurveyContextProvider@http://localhost:7107/static/js/bundle.js:20004:85
Route@http://localhost:7107/static/js/bundle.js:274891:29
CheckIfUserHasProjectRole@http://localhost:7107/static/js/bundle.js:16904:7
WaitForProjectParticipantInfo@http://localhost:7107/static/js/bundle.js:16749:84
ProjectRoleRouteGuard@http://localhost:7107/static/js/bundle.js:16579:7
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
ProjectsLayout@http://localhost:7107/static/js/bundle.js:60594:19
Route@http://localhost:7107/static/js/bundle.js:274891:29
RouteWithTitle@http://localhost:7107/static/js/bundle.js:63199:7
ProjectContextProvider@http://localhost:7107/static/js/bundle.js:19881:85
ProjectAuthStateContextProvider@http://localhost:7107/static/js/bundle.js:19729:85
Route@http://localhost:7107/static/js/bundle.js:274891:29
RouteWithTitle@http://localhost:7107/static/js/bundle.js:63199:7
Switch@http://localhost:7107/static/js/bundle.js:275060:29
ProjectsRouter
CodesContextProvider@http://localhost:7107/static/js/bundle.js:19055:85
Route@http://localhost:7107/static/js/bundle.js:274891:29
CheckIfAuthenticatedUser@http://localhost:7107/static/js/bundle.js:16779:56
WaitForKeycloakToLoadUserInfo@http://localhost:7107/static/js/bundle.js:16715:39
CheckForKeycloakAuthenticated@http://localhost:7107/static/js/bundle.js:16686:56
AuthenticatedRouteGuard@http://localhost:7107/static/js/bundle.js:16616:7
main
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
DialogContextProvider@http://localhost:7107/static/js/bundle.js:19367:98
div
./node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:65443:66
Box@http://localhost:7107/static/js/bundle.js:118389:72
BaseLayout@http://localhost:7107/static/js/bundle.js:60464:19
Route@http://localhost:7107/static/js/bundle.js:274891:29
RouteWithTitle@http://localhost:7107/static/js/bundle.js:63199:7
Switch@http://localhost:7107/static/js/bundle.js:275060:29
AppRouter@http://localhost:7107/static/js/bundle.js:213:82
Router@http://localhost:7107/static/js/bundle.js:274563:30
BrowserRouter@http://localhost:7107/static/js/bundle.js:274078:35
AuthStateContextProvider@http://localhost:7107/static/js/bundle.js:18966:95
KeycloakProvider@http://localhost:7107/static/js/bundle.js:161152:47
ConfigContextProvider@http://localhost:7107/static/js/bundle.js:19216:78
ThemeProvider@http://localhost:7107/static/js/bundle.js:115989:7
ThemeProvider@http://localhost:7107/static/js/bundle.js:117623:7
ThemeProvider@http://localhost:7107/static/js/bundle.js:114273:9
App
```

(tested on my local)

**Comment by NickPhura** [comment]
Also, I'm getting a 503 error from the `http://localhost:6107/api/project/1/survey/1/deployments` endpoint when the survey page loads.

(tested on my local)

File: api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/deployments.ts]: Can make it required too, since its always provided.

**Review by al-rosenthal — DISMISSED** [review]
I'm seeing a 500 on the deployments api call when a survey loads. Am I missing some sort of setup for testing this?

**Review by GrahamS-Quartech — APPROVED** [review]

**Review by NickPhura — APPROVED** [review]

**Pull request merged**